File: shortcut/src/main/java/com/muugi/shortcut/core/IntentSenderHelper.java
package com.muugi.shortcut.core;

import android.app.PendingIntent;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.IntentSender;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by ZP on 2019/3/3.
 */
public class IntentSenderHelper {

    private static final String TAG = "IntentSenderHelper";

    public static IntentSender getDefaultIntentSender(@NonNull Context context, @NonNull String action) {
        return PendingIntent.getBroadcast(context, 0, new Intent(action),
                PendingIntent.FLAG_ONE_SHOT).getIntentSender();
    }


    public static IntentSender getDefaultIntentSender(@NonNull Context context, @NonNull String action, Class<?> clz, @Nullable Bundle bundle) {
        Intent intent = new Intent(action);
        intent.setComponent(new ComponentName(context, clz));
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        return PendingIntent.getBroadcast(context, 0, intent,
                PendingIntent.FLAG_ONE_SHOT).getIntentSender();
    }
}

File: shortcut/src/main/java/com/muugi/shortcut/broadcast/IntentSenderHelper.kt
package com.muugi.shortcut.broadcast

import android.app.PendingIntent
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.IntentSender
import android.os.Bundle

/**
 * Created by ZP on 2019/3/3.
 */
object IntentSenderHelper {
    fun getDefaultIntentSender(context: Context, action: String): IntentSender {
        return PendingIntent.getBroadcast(context, 0, Intent(action),
                PendingIntent.FLAG_ONE_SHOT).intentSender
    }

    fun getDefaultIntentSender(context: Context, action: String, clz: Class<*>, bundle: Bundle?): IntentSender {
        val intent = Intent(action)
        intent.component = ComponentName(context, clz)
        if (bundle != null) {
            intent.putExtras(bundle)
        }
        return PendingIntent.getBroadcast(context, 0, intent,
                PendingIntent.FLAG_ONE_SHOT).intentSender
    }
}